### DISCUSSION TITLE
tiptap remove div and class

### DISCUSSION THREAD

**Discussion by myalban** [ueberdosis/tiptap#3792] [General Questions] [answered]
Hi,

I have this sample text  :  

`<div class="flex flex-row"><p class="text-red-600">text red </p> <i class="icon-up"></i></div>`

but getHTML() output  : 

`<p class="text-red-600">text redd</p>`

I'm pretty sure that there is something to setup but can't find it.

**Comment by bschmalz** [comment]
I'm running into this issue and notice this was never resolved. Anyone have a fix for this?

**Reply by nperez0111** [reply, marked as answer]
Tiptap strips any HTML that does not adhere to it's strict schema, read about it on our guide: https://tiptap.dev/docs/editor/core-concepts/schema

So if there is HTML that does not conform to the schema (i.e. it does not match a node or mark) it will be stripped from the HTML. This is so that the editor can handle the HTML in a known format otherwise it would not know what to do with the "extra" elements.

**Reply by bschmalz** [reply]
Is there an example of a basic schema that provides support for divs so that they aren't removed from the html?

**Reply by nperez0111** [reply]
I don't think you read the guide if you are asking that question…

You'd need a node with a parseHTML that accepts divs but I would warn you that this is probably not what you actually want.

You did not give enough information to figure out what you actually wanted. This is an XY problem

Tiptap is not an arbitrary HTML editor do not treat it like one thinking that it will work
